### PR TITLE
clarify loop inserting func into vtbl[]

### PR DESCRIPTION
**Pull request by WalterBright**
This was a weird forward then backward loop, with a confusing comment. Converted it to a conventional loop, improved the comment, and added a debug check that clarifies what it does.

(In general, loops are more comprehensible if they go forward in a conventional, predictable manner.)

### CODE REVIEW

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -3311,24 +3311,31 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                     funcdecl.introducing = 1;
                     if (cd.classKind == ClassKind.cpp && Target.reverseCppOverloads)
                     {
-                        // with dmc, overloaded functions are grouped and in reverse order
+                        /* Overloaded functions with same name are grouped and in reverse order.
```
Comment (thewilsonator) [inline src/dmd/dsymbolsem.d]: What about DMC? the previous comment mentions it. Does it no longer apply to just DMC?

**Review by thewilsonator — APPROVED** [review]
Otherwise LGTM

File: src/dmd/dsymbolsem.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/dsymbolsem.d]: I figured the `Target.reverseCppOverload` on line 3312 covered that.

Comment (thewilsonator) [reply src/dmd/dsymbolsem.d]: OK.

**Comment by thewilsonator** [comment]
IDK whats with appveyor, try giving it a kick?

**Comment by rainers** [comment]
> IDK whats with appveyor, try giving it a kick?

Considering that the test checks vtable layout it's unlikely to be spurious. The same test fails for the auto-tester.

File: src/dmd/dsymbolsem.d [hunk 2]
```diff
@@ -3311,24 +3311,32 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
...
-                                break;
+                                found = true;
+                                ++s.isFuncDeclaration().vtblIndex;
                             }
                         }
-                        // shift all existing functions back
-                        for (size_t i = cd.vtbl.dim; i > funcdecl.vtblIndex; i--)
+                        cd.vtbl.insert(funcdecl.vtblIndex, funcdecl);
+
+                        debug foreach (const i, s; cd.vtbl)
```
Comment (look-at-me) [inline src/dmd/dsymbolsem.d]: @WalterBright 

Why was this put in a debug block? This assert is failing on windows 64-bit but none of the tests use a debug build.

**Pull request merged**